File: backend/src/main/java/nextpos/app/nextpos/model/enums/ShipmentStatus.java
package nextpos.app.nextpos.model.enums;

public enum ShipmentStatus {
    PENDING,       // Shipping is created but not yet processed
    PROCESSING,    // Shipping is being processed/prepared
    SHIPPED,       // Shipment is dispatched
    IN_TRANSIT,    // Shipment is on the way
    DELIVERED,     // Shipment delivered to destination
    CANCELLED,     // Shipment was cancelled
    RETURNED       // Shipment was returned
}

File: frontend/src/enums/shipmentStatus.ts
export enum ShipmentStatus {
  PENDING = "PENDING",
  PROCESSING = "PROCESSING",
  SHIPPED = "SHIPPED",
  IN_TRANSIT = "IN_TRANSIT",
  DELIVERED = "DELIVERED",
  RETURNED = "RETURNED",
}

export const ShipmentStatusLabels: Record<ShipmentStatus, string> = {
  [ShipmentStatus.PENDING]: "Pending",
  [ShipmentStatus.PROCESSING]: "Processing",
  [ShipmentStatus.SHIPPED]: "Shipped",
  [ShipmentStatus.IN_TRANSIT]: "In Transit",
  [ShipmentStatus.DELIVERED]: "Delivered",
  [ShipmentStatus.RETURNED]: "Returned",
};
